### PR TITLE
Refactor data module to wrapper around data frame

### PR DESCRIPTION
**Pull request by zhengp0**
* Add `Column` classes to enforce validation and default values
* Wrap `MRData` around `DataFrame` for easy accessing data and preserve row relations

### CODE REVIEW

File: src/mrtool/core/data.py [hunk 1]
```diff
@@ -5,362 +5,227 @@
...
+            raise TypeError("df must be DataFrame or None.")
+
+    def _check_df_values(self, df: DataFrame):
+        if self.name not in df.columns:
+            self._set_default_values(df)
+        if any(df[self.name].isna()):
+            raise ValueError(f"Column({self.name}) values cannot contain NaN.")
+
+    def _set_default_values(self, df: DataFrame):
+        raise ValueError(f"df must contain column {self.name}.")
```
Comment (jiaweih) [inline src/mrtool/core/data.py:37]: So this is actually not setting values?

Comment (zhengp0) [reply src/mrtool/core/data.py:37]: No for the default `Column` class it doesn't, but the child class like `SEColumn` and `KeyColumn` it does :)

File: src/mrtool/core/data.py [hunk 2]
```diff
@@ -5,362 +5,227 @@
...
-        """If the data has provided study_id
+    @property
+    def shape(self):
+        self._assert_not_empty()
+        return self.df.shape
 
-        Args:
-            studies Union[List[Any], Any]:
-                List of studies or one study.
+    def sort_values(self, by: Union[str, List[str]]):
```
Comment (jiaweih) [inline src/mrtool/core/data.py:206]: Do we need an option argument of `ascending=True`?

File: src/mrtool/core/data.py [hunk 3]
```diff
@@ -5,362 +5,227 @@
...
+        return dict(**required_cols, other_cols=other_cols)
+
+    @property
+    def df(self) -> DataFrame:
+        return self._df
+
+    @df.setter
+    def df(self, df: DataFrame):
+        for col in self.cols:
+            col.df = df
```
Comment (jiaweih) [inline src/mrtool/core/data.py:190]: We might only need the df column corresponding to `col`, but i guess it's also fine to attach the whole df

File: src/mrtool/core/data.py (hunk 2 above)
Comment (zhengp0) [reply src/mrtool/core/data.py]: Just changed it :) Thanks very much! Now it has all options.

File: src/mrtool/core/data.py (hunk 3 above)
Comment (zhengp0) [reply src/mrtool/core/data.py:190]: yes! that is one of the decision, I think it might be good to just preserve the whole dataframe :)

**Pull request merged**